src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -77,23 +74,13 @@ public LinterAnalyzer()
 
         internal IEnumerable<IDiagnostic> Analyze(SemanticModel semanticModel, ConfigHelper? overrideConfig = default)
```

Comment:
Now that we have `activeConfigHelper` set via the constructor, do we need to keep this overload which accepts `overrideConfig`? If so, do you know which code paths require it? It's always felt like an anti-pattern, as it negates the ability to cache and reuse semantic models for diagnostics.

Reply:
It was used in tests. I removed passing overrideConfig as part of this commit:
https://github.com/Azure/bicep/commit/1e2de9b409ea393d9b1c034e9f615d5cf1bc3ca6